packages/server/src/db/index.js

```diff
@@ -1040,12 +1040,112 @@ async function inTenant(tenantId, agencyIds) {
     return result.same_tenant === true;
 }
 
+/**
+ * Creates and saves a new saved search, given a name, agency ID, user ID, and criteria
+ *  interface SavedSearch {
+    id?: number
+    name: string
+    agencyId: number
+    userId: number
+    criteria: string
+    createdAt?: string
+   }
+ * @param  SavedSearch           searchItem
+ * @return Promise<SavedSearch>
+ * */
+async function createSavedSearch(searchItem) {
+    const response = await knex
+        .insert({
+            name: searchItem.name,
+            agency_id: searchItem.agencyId,
+            created_by: searchItem.userId,
+            criteria: searchItem.criteria,
+        })
+        .into('grants_saved_searches')
+        .returning('*');
+
+    return {
+        id: response[0].id,
+        name: response[0].name,
+        agencyId: response[0].agency_id,
+        createdBy: response[0].created_by,
+        criteria: response[0].criteria,
+        createdAt: new Date(response[0].created_at).toISOString(),
+    };
+}
+
+/**
+ * Retrieves saved searches, given an agency ID, result limit, and result offset
+ * @param  int              userId
+ * @param  int              agencyId
+ * @param  IPaginateParams  paginationParams
+ * @return Promise<boolean>
+ * */
+async function getSavedSearches(userId, agencyId, paginationParams) {
+    const response = await knex('grants_saved_searches')
+        .where('created_by', userId)
+        .andWhere('agency_id', agencyId)
+        .orderBy('id')
+        .paginate(paginationParams);
+
+    response.data = response.data.map((r) => ({
+        id: r.id,
+        name: r.name,
+        agencyId: r.agency_id,
+        createdBy: r.created_by,
+        criteria: r.criteria,
+        createdAt: new Date(r.created_at).toISOString(),
+    }));
+
+    return response;
+}
+
+/**
+ * Get Saved Search with ID
+ * @param  int              id
+ * @param  int              agencyId
+ * @return Promise<boolean>
```

Comment:
Is this return parameter boolean, or the row | null?

Reply:
good point, I updated